File: java/SelectionSort.java
// time complexity; O(n^2)
// space complexity; O(1)

public class SelectionSort
  {
    public static void selectionSort(int[] arr)
    {
      int n = arr.length;
      // one by one move boundary of unsorted subarray 
      for(int i = 0; i < n - 1; ++i)
        {
          // find min element in unsorted array
          int min_idx = i;
          for(int j = i + 1; j < n; j ++)
            if(arr[j] < arr[min_idx])
              min_idx = j;
          // swap found min element with first element
          int temp = arr[min_idx];
          arr[min_idx] = arr[i];
          arr[i] = temp;
        }
    }
  }

File: python/selectionSort.py
# time complexity; O(n^2)
# space complexity: O(1)

def selection_sort(arr):
  n = len(arr)
  for i in range(n):
    min_idx = i
    for j in range(i + 1, n):
      if arr[min_idx] > arr[j]:
        if arr[min_idx] > arr[j]:
          min_idx = j
    arr[i], arr[min_idx] = arr[min_idx], arr[i]

